ml/src/svm.cpp

```diff
@@ -2351,8 +2351,14 @@ svm_group_classes(const svm_problem* prob,
     if (j == nr_class) {
       if (nr_class == max_nr_class) {
         max_nr_class *= 2;
-        label = static_cast<int*>(realloc(label, max_nr_class * sizeof(int)));
-        count = static_cast<int*>(realloc(count, max_nr_class * sizeof(int)));
+        int* label_temp = static_cast<int*>(realloc(label, max_nr_class * sizeof(int)));
+        if (label_temp) {
+          label = label_temp;
+        }
+        int* count_temp = static_cast<int*>(realloc(count, max_nr_class * sizeof(int)));
+        if (count_temp) {
+          count = count_temp;
+        }
```

Comment:
Move the changes in this file to a separate PR. It needs more fundamental changes. It should be using vectors instead of handling reallocs manually.

Reply:
I have reverted the changes for these now except freeing `model->scaling`